lib/utils/certs.go

```diff
@@ -161,15 +161,83 @@ func ParsePrivateKeyDER(der []byte) (crypto.Signer, error) {
 // chain from leaf to root is valid. This ensures that clients (web browsers
 // and CLI) won't have problem validating the chain.
 func VerifyCertificateChain(certificateBytes []byte) error {
```

Comment:
I'm not sure I like that VerifyCertifivcatChain should accept the self signed certificates as valid. I think this is a special case should be handled on the higher level may be?